### PR TITLE
Sesson submit log should be overwritten

### PR DESCRIPTION
**Pull request by ulysses-you**
We will create and write a log during submit a new session (like spark-submit) but we don't delete or overwrite it. This PR aims to add a time config to decide if we need to overwrite a session submit log.

The log path template is `KYUUBI_WORD_DIR/user/$engine-log.$index`. We will increase the `index` automatically if a user create some session concurrently.

After this PR. The log path template is `KYUUBI_WORD_DIR/user/$engine.log.$index`.

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @yaooqinn how about this implemention

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -355,6 +355,15 @@ object KyuubiConf {
     .transform(_.toLowerCase(Locale.ROOT))
     .createWithDefault(SaslQOP.AUTH.toString)
 
+  val SESSION_SUBMIT_LOG_RETAIN_MILLIS: ConfigEntry[Long] =
+    buildConf("session.submit.log.retain.millis")
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:359]: can we just build the conf with `session.engine.log.timeout`

as we have 

session.engine.idle.timeout
session.engine.login.timeout
session.engine.initialize.timeout

etc

It's much easier to group them in the documentation and in the user's mind.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -355,6 +355,15 @@ object KyuubiConf {
...
+      .doc("If we use Spark as the engine then the session submit log is the console output of " +
+        "spark-submit. We will retain the session submit log until over the config value.")
+      .version("1.1.0")
+      .longConf
+      .checkValue(_ > 0, "must be positive number")
+      .createWithDefault(7 * 24 * 3600 * 1000L)
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:365]: Duration.ofDays(1).toMillis

**Comment by yaooqinn** [comment]
we need to run the tests locally, which helps to generate the doc if we add a new kyuubi config

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (ulysses-you) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: yeah see it.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 3]
```diff
@@ -355,6 +355,15 @@ object KyuubiConf {
     .transform(_.toLowerCase(Locale.ROOT))
     .createWithDefault(SaslQOP.AUTH.toString)
 
+  val SESSION_SUBMIT_LOG_RETAIN_MILLIS: ConfigEntry[Long] =
+    buildConf("session.submit.log.retain.millis")
+      .doc("If we use Spark as the engine then the session submit log is the console output of " +
+        "spark-submit. We will retain the session submit log until over the config value.")
+      .version("1.1.0")
+      .longConf
```
[the remark this answers is not in the document]
Comment (ulysses-you) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: Updated

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 2 above)
Comment (ulysses-you) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: Done

**Comment by yaooqinn** [comment]
thanks merged to master for 1.1

**Comment by yaooqinn** [comment]
```
- log capture should release after close
- sub process log should be overwritten *** FAILED ***
  Array(/home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.9, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.7, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.3, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.8, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.5, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.1, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.2, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.4, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.0, /home/runner/work/kyuubi/kyuubi/kyuubi-main/target/tmp/fake2420230266546989920/kyuubi-spark-sql-engine.log.6) had length 10 instead of expected length 11 (SparkProcessBuilderSuite.scala:150)
``` 

the test seems flaky?

**Comment by ulysses-you** [comment]
> the test seems flaky?

Let me take a close look.

**Pull request closed without merging**